.github/workflows/pull_request.workflow.yml

```diff
@@ -0,0 +1,114 @@
+name: Pull request checks
+
+on: [pull_request]
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - uses: actions/setup-node@v1.4.4
+        with:
+          node-version: "12"
+      - uses: ./.github/actions/es-lint
+
+  unit-tests:
+     name: Unit Tests
+     runs-on: ubuntu-latest
+     steps:
+       - uses: actions/checkout@v2
+       - name: Cache node modules
+         uses: actions/cache@v2
+         env:
+           cache-name: cache-node-modules
+         with:
+           path: ~/.npm
+           key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+           restore-keys: |
+             ${{ runner.os }}-build-${{ env.cache-name }}-
+             ${{ runner.os }}-build-
+             ${{ runner.os }}-
+       - uses: actions/setup-node@v1.4.4
+         with:
+           node-version: "12"
+       - uses: ./.github/actions/unit-tests
+
+  integration-tests:
+     name: Integration Tests
+     runs-on: ubuntu-latest
+     steps:
+       - uses: actions/checkout@v2
+       - name: Cache node modules
+         uses: actions/cache@v2
+         env:
+           cache-name: cache-node-modules
+         with:
+           path: ~/.npm
+           key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+           restore-keys: |
+             ${{ runner.os }}-build-${{ env.cache-name }}-
+             ${{ runner.os }}-build-
+             ${{ runner.os }}-
+       - uses: actions/setup-node@v1.4.4
+         with:
+           node-version: "12"
+       - uses: ./.github/actions/integration-tests
+         env:
+          CYPRESS_RECORD_KEY_DOC: ${{ secrets.CYPRESS_RECORD_KEY_DOC }}
+
+  docs-tests:
+     name: Documentation Snippet Tests
+     runs-on: ubuntu-latest
+     steps:
+       - uses: actions/checkout@v2
+       - name: Cache node modules
+         uses: actions/cache@v2
+         env:
+           cache-name: cache-node-modules
+         with:
+           path: ~/.npm
+           key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+           restore-keys: |
+             ${{ runner.os }}-build-${{ env.cache-name }}-
+             ${{ runner.os }}-build-
+             ${{ runner.os }}-
+       - uses: actions/setup-node@v1.4.4
+         with:
+           node-version: "12"
+       - uses: ./.github/actions/snippet-tests
+         env:
```

Comment:
Do not use environnement variables for your custom actions prefer the inputs/outputs system directly from your action file